internal/controllers/machineset/machineset_controller.go

```diff
@@ -987,6 +1047,22 @@ func (r *Reconciler) reconcileUnhealthyMachines(ctx context.Context, cluster *cl
 		return ctrl.Result{}, nil
 	}
 
+	// Sort the machines from newest to oldest.
```

Comment:
nit
```suggestion
	// Sort the machines from newest to oldest.
	// Note: We are trying to remediate machines failing to come up first because there is a chance that they won't host any workload yet (minimize disruption).
```